clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -440,6 +440,7 @@ private void handleCompletedReceives(List<ClientResponse> responses, long now) {
             correlate(req.request().header(), header);
             if (!metadataUpdater.maybeHandleCompletedReceive(req, now, body))
                 responses.add(new ClientResponse(req, now, false, body));
+            inFlightRequests.completeNext(source);
```

Comment:
When you see this issue, what happens in the client? Does it get an exception? We probably want the client to get an exception in this case. Also, if this is due to a broker sending a wrong response, we likely should fix the cause than the symptom, right? Do we know the cause yet?